pkg/descheduler/pod/pods.go

```diff
@@ -257,3 +258,20 @@ func SortPodsBasedOnAge(pods []*v1.Pod) {
 		return pods[i].CreationTimestamp.Before(&pods[j].CreationTimestamp)
 	})
 }
+
+// GetOwnerReplicasCount returns the count of pods that match the owner ref of the given pod
+func GetOwnerReplicasCount(pods []*v1.Pod, pod *v1.Pod) uint {
+	var count uint
+	podOwnerRefs := OwnerRef(pod)
+	for _, otherPod := range pods {
+		otherOwnerRefs := OwnerRef(otherPod)
+
+		if slices.EqualFunc(podOwnerRefs, otherOwnerRefs, func(l, r metav1.OwnerReference) bool {
```

Comment:
How often it is to have a pod with two or more owners? Also, if a pod A is owned by owner X and owner Y where X owns 5 replicas and Y owns 6 replicas. Will it mean that when a set of pods is owned by both X and Y the right number of replicas is 5? Or, is it max(5,6)? I'd vote for ignoring any pod that has two or more owners as the first iteration of the implementation. If there's a use case where it make sense to have 2 or more owners we can analyse it per case and alter the implementation. WDYT?

Reply:
`slices.EqualFunc` will return false if the len is not the same. So in the example you gave, it will be 5.

Reply:
I just pushed up a commit, let me know if that's what you had in mind

Reply:
Yes, that's what I meant :) Looks great. Thank you.

Reply:
awesome! thanks for the hint. just squashed commits